lib/streamlit/elements/widgets/data_editor.py

```diff
@@ -817,15 +817,31 @@ def data_editor(
 
         # Convert the user provided column config into the frontend compatible format:
         column_config_mapping = process_config_mapping(column_config)
-        apply_data_specific_configs(
-            column_config_mapping, data_df, data_format, check_arrow_compatibility=True
-        )
+
+        # Deactivate editing for columns that are not compatible with arrow
+        for column_name, column_data in data_df.items():
```

Comment:
do you think we should have some measurements in the future for this kind of iterations? Or do you know that these iterations are very cheap?
Same with all of the checks, like `is_colum_type_arrow_incompatible`?

Reply:
I think this operation is mostly a pointer iteration and quite cheap. But doing some measurements/profiling around this to get a better understanding sounds reasonable. I will write that down and try to look into it later the sprint.